src/main/scala/treecanopy/Main.scala

```diff
@@ -3,70 +3,130 @@ package treecanopy
 import geotrellis.proj4._
 import geotrellis.raster._
 import geotrellis.raster.render._
-import geotrellis.raster.render.png._
 import geotrellis.raster.resample._
 import geotrellis.spark._
-import geotrellis.spark.etl.Etl
-import geotrellis.spark.io._
 import geotrellis.spark.io.s3._
 import geotrellis.spark.pyramid.Pyramid
 import geotrellis.spark.render._
 import geotrellis.spark.tiling._
 import geotrellis.spark.util.SparkUtils
 import geotrellis.vector._
-import geotrellis.vector.io._
 
 import com.amazonaws.services.s3.model._
-import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.mapreduce.Job
 import org.apache.spark._
 import org.apache.spark.rdd.RDD
 
+
 object Ingest {
+  val sourceBucket = "azavea-datahub"
+  val statewideLayerName = "raw/treecanopy-2006-2008-pennysylvania-albers"
+  val localLayerNames = List(
+    "raw/treecanopy-pennsylvania-local/abingtons",
+    "raw/treecanopy-pennsylvania-local/alleghenyCounty",
+    "raw/treecanopy-pennsylvania-local/harrisburg",
+    "raw/treecanopy-pennsylvania-local/lancasterCity",
+    "raw/treecanopy-pennsylvania-local/montgomeryCounty",
+    "raw/treecanopy-pennsylvania-local/stateCollege"
+  )
+  val outputPath = "s3://com.azavea.datahub.tms/{name}/{z}/{x}/{y}.png"
+  val outputName = "treecanopy-2006-2008-pa-merged"
+
+  val tileSize = 256
   val targetLayoutScheme =
-    ZoomedLayoutScheme(WebMercator, 512)
+    ZoomedLayoutScheme(WebMercator, tileSize)
 
+  // Map 1 to the canopy color and everything else to 0
   val colorMap =
     ColorMap(
       Map(
         0 -> 0x00000000,
-        1 -> 0x139A68FF
+        1 -> 0x139A68FF,
+        Int.MaxValue -> 0x00000000
       )
     )
 
   def main(args: Array[String]): Unit = {
     implicit val sc = SparkUtils.createSparkContext("Tree Canopy ETL", new SparkConf(true))
 
+    val maxLocalZoom = 19  // highest zoom level of local datasets
+    val partitioner = new HashPartitioner(5000)
```

Comment:
You should use the same partitioner for all of the tiling of the rasters, so by the time you are combining them by key, all the tiles with the same keys live on the same machines